number_theory/numeric_unittest.cpp

```diff
@@ -81,5 +81,39 @@ TEST(NumericTest, Pow) {
   EXPECT_DOUBLE_EQ(pow(2, 1.1), std::pow(2, 1.1));
 }
 
+template <typename T>
+void test_iroot() {
+  // test small numbers
+  EXPECT_EQ(iroot(T(120), 2), 10);
+  EXPECT_EQ(iroot(T(125), 3), 5);
+  EXPECT_EQ(iroot(T(0), 1), 0);
+  if (std::numeric_limits<T>::is_signed) {
+    EXPECT_EQ(iroot(T(-125), 3), -5);
+    EXPECT_EQ(iroot(T(-30), 3), -3);
+  }
+
+  // test exceptions
```

Comment:
I think we should also test `std::invalid_argument` exception with a negative `n` case to cover all the execution paths in the tests.

Reply:
I think we don't need to test undocumented exceptions? Like we didn't test invalid_argument in inverse_mod.

Reply:
I think we should test it. I did not realize that I missed it during tests. The point is to test all execution paths which is called whitebox testing. The reason we do not document the exception is that it contradicts with the documented conditions. Thus, documenting a contradicting thing makes it look like it is part of the API. It is like we are suggesting them that it should be used that way. In reality, for example, the mod should never be negative. It is not a feature that we throw exception when they provide negative modulus. However, we still need to test it to make sure that it does not break.